GAD 6 - Verbessertes Sortieren/tests/simplesort/TesterDualQuicksortResult.java

```diff
@@ -50,27 +57,36 @@ public void logPartialArray(int[] array, int from, int to) {
     }
 
     private void analyseLogsOfDualPivot(int[] endResult) {
-        if (startResult == null || endResult == null) {
+        if (numbersAtStart == null || endResult == null) {
             throw new IllegalArgumentException("Array: " + Arrays.toString(endResult) + "\n" + "\nIn range: " + indicesOfLog[0] +
                     " to " + indicesOfLog[5] + "\n" + "\nLogged partial Arrays without logging startResult or endResult");
         }
 
-        Arrays.sort(startResult);
+        int[] numbersAtStartPartial = Arrays.copyOfRange(numbersAtStart, indicesOfLog[0], indicesOfLog[5] + 1);
+        Arrays.sort(numbersAtStartPartial);
         int[] copyEndResult = Arrays.copyOfRange(endResult, indicesOfLog[0], indicesOfLog[5] + 1);
         Arrays.sort(copyEndResult);
 
-        assertEquals(startResult.length, copyEndResult.length, "Array: " + Arrays.toString(endResult) + "\nIn range: "
+        assertEquals(numbersAtStartPartial.length, copyEndResult.length, "Array: " + Arrays.toString(endResult) + "\nIn range: "
                 + indicesOfLog[0] + " to " + indicesOfLog[5] + "\nThe arrays before and after the sort dont have the same length anymore, " +
                 "you probably changed the to or from pointer");
 
-
-        assertArrayEquals(copyEndResult, startResult, "Array: " + Arrays.toString(endResult) + "\nIn range: " +
+        assertArrayEquals(copyEndResult, numbersAtStartPartial, "Array: " + Arrays.toString(endResult) + "\nIn range: " +
                 indicesOfLog[0] + " to " + indicesOfLog[5] + "\nThe arrays before and after the sort have different numbers");
 
+        int[] pivots = pivotFinder.findPivot(numbersAtStart, indicesOfLog[0], indicesOfLog[5]);
+        int valuePivotStartFirst = numbersAtStart[pivots[0]];
+        int valuePivotStartSecond = numbersAtStart[pivots[1]];
 
         int pivotOne = endResult[indicesOfLog[1] + 1];
         int pivotTwo = endResult[indicesOfLog[3] + 1];
 
+        if (!(valuePivotStartFirst == pivotOne && valuePivotStartSecond == pivotTwo) &&
+                !(valuePivotStartFirst == pivotTwo && valuePivotStartSecond == pivotOne))
+            fail("Array: " + Arrays.toString(endResult) +
+                    "\nIn range: " + indicesOfLog[0] + " to " + indicesOfLog[5] +
+                    "\nThe pivots of the pivot finder do not correspond to the log Pivots");
+
         if (pivotOne > pivotTwo) fail("Array: " + Arrays.toString(endResult) +
```

Comment:
Please add a line break or opening & closing braces around the inner `if` block here

Reply:
(This would also apply to the other `if` statement below, especially the line break suggestion.)